neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -125,8 +125,10 @@ public override int GetHashCode()
 
         public UInt160[] GetScriptHashesForVerifying(Snapshot snapshot)
         {
-            var hashes = new HashSet<UInt160> { Sender };
-            hashes.UnionWith(Cosigners.Select(p => p.Account));
+            if (Cosigners.Length == 0) return new UInt160[] { Sender };
+
+            var hashes = new HashSet<UInt160>(Cosigners.Select(p => p.Account));
+            hashes.UnionWith(new HashSet<UInt160> { Sender });
             return hashes.OrderBy(p => p).ToArray();
```

Comment:
Maybe we can avoid the order.

Reply:
I think that we don't need the order now, Isn't it @erikzhang ?

```suggestion
            return hashes.ToArray();
```

Reply:
Yes, we need the order.